src/qonlinetranslator.cpp

```diff
@@ -269,6 +269,42 @@ bool QOnlineTranslator::isRunning() const
     return m_stateMachine->isRunning();
 }
 
+QString QOnlineTranslator::toJson() const
+{
+    QJsonObject translationOptions;
+    for (auto it = m_translationOptions.begin(); it != m_translationOptions.end(); ++it) {
+        QJsonArray arr;
+        for (const QOption &option : it.value()) {
+            arr.append(option.toJson());
+        }
+        translationOptions.insert(it.key(), arr);
+    }
+
+    QJsonObject examples;
+    for (auto it = m_examples.begin(); it != m_examples.end(); ++it) {
+        QJsonArray arr;
+        for (const QExample &example : it.value()) {
+            arr.append(example.toJson());
+        }
+        examples.insert(it.key(), arr);
+    }
+
+    QJsonObject object
+    {
+        {"source", m_source},
+        {"sourceTranslit", m_sourceTranslit},
+        {"sourceTranscription", m_sourceTranscription},
+        {"translation", m_translation},
+        {"translationTranslit", m_translationTranslit},
+        {"errorString", m_errorString},
+        {"translationOptions", qMove(translationOptions)},
+        {"examples", qMove(examples)}
+    };
+
+    QJsonDocument doc = QJsonDocument(object);
+    return QString(doc.toJson(QJsonDocument::Indented));
```

Comment:
Could you return `QJsonDocument` directly (change function return type)? It would be more flexible for external use.

```suggestion
    return QJsonDocument(object);
```